functions/helloworld/main.py

```diff
@@ -80,19 +80,22 @@ def hello_http(request):
 
 
 # [START functions_helloworld_pubsub]
-def hello_pubsub(data, context):
+def hello_pubsub(event, context):
     """Background Cloud Function to be triggered by Pub/Sub.
     Args:
-         data (dict): The dictionary with data specific to this type of event.
+         event (dict):  The dictionary with data specific to this type of
+         event. The `data` field contains the PubsubMessage message. The
+         `attributes` field will contain custom attributes if there are any.
          context (google.cloud.functions.Context): The Cloud Functions event
-         metadata.
+         metadata. The `event_id` field contains the Pub/Sub message ID. The
+         `timestamp` field contains the publish time.
     """
     import base64
 
-    if 'data' in data:
-        name = base64.b64decode(data['data']).decode('utf-8')
+    if 'data' in event:
+        name = base64.b64decode(event['data']).decode('utf-8')
     else:
-        name = 'World'
+        name = 'World! Something went wrong'
```

Comment:
Is "something went wrong" always true in the case where there's no data, or is it sometimes just a consequence that the user didn't add any? If something really is wrong, can we be more specific for the user's benefit?

Reply:
The goal there is to print something other than "World!" Otherwise the output will always be "Hello World!"

Reply:
If something really did go wrong, maybe raise an exception?

Reply:
If that's the case - maybe we should remove this `if-else` logic from this sample? (Though the current logic does allow non-Pub/Sub triggers to invoke the function without issue.)

IMO, we shouldn't be documenting how to handle (relatively rare) GCP-side system failures (on this page at least).